src/main/java/uk/gov/justice/digital/delius/service/NsiService.java

```diff
@@ -45,4 +45,16 @@ public NsiService(final NsiRepository nsiRepository, final ConvictionService con
         return nsiRepository.findById(nsiId)
                 .map(NsiTransformer::nsiOf);
     }
+
+    @Transactional(readOnly = true)
+    public NsiWrapper getNsiByCodesForOffenderActiveConvictions(final Long offenderId, final Collection<String> nsiCodes) {
+
+        return new NsiWrapper(
+                nsiRepository.findByOffenderIdForActiveEvents(offenderId)
+                        .stream()
+                        .filter(nsi -> !nsi.isSoftDeleted())
```

Comment:
Maybe add soft_deleted to query rather than filter here?

Reply:
yeah - was going with the convention here, happy to change